microsoft/testsuites/sysbench/sysbenchperf.py

```diff
@@ -0,0 +1,406 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import json
+import re
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict, List
+
+from assertpy.assertpy import fail
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+)
+from lisa.messages import TestStatus, send_sub_test_result_message
+from lisa.testsuite import TestResult
+from lisa.tools import Sysbench
+from lisa.util.process import ExecutableResult
+
+
+@dataclass
+class SysbenchTestResult:
+    name: str = ""
+    status: TestStatus = TestStatus.QUEUED
+    message: str = ""
+
+
+@TestSuiteMetadata(
+    area="sysbench",
+    category="performance",
+    description="""
+    This test suite is for executing the sysbench tests
+    """,
+)
+class SysbenchPerfTestSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            Runs Sysbench test for cpu
+        """,
+        priority=3,
+    )
+    def verify_sysbench_cpu_perf(
+        self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        variables: Dict[str, Any],
+    ) -> None:
+        subtest_name: str = "default_cpu_perf_check"
+        res: SysbenchTestResult = SysbenchTestResult()
+
+        output: ExecutableResult = node.tools[Sysbench].run_cpu_perf()
+
+        if output.exit_code != 0:
```

Comment:
In LISA, it's done in one line, including mark the test case fails. Please refer to how other test cases implemented.

`output.assert_exit_code(expected_exit_code=0, message="...")`

Reply:
This would not let push the failed message for subtest and testcase would throw the error from there.

We already have assert_that on status. exit code check if to determine which message should be send as subtest result.

We are using subtest here because, as of now, we have default run for cpu performance using sysbench, but in future we might run the same with different parameters. In that case, subtest (having different parameters) would make more sense.